rust-connectors/sources/mqtt/src/error.rs

```diff
@@ -16,12 +17,16 @@ pub enum MqttConnectorError {
     MqttConnection(#[from] MqttConnectionError),
     #[error("Mqtt client error: `{0:#?}`.")]
     MqttClient(#[from] MqttClientError),
+    #[error("Mqtt OptionError: `{0:#?}`.")]
```

Comment:
I think that we should use the display representation in most of these errors

Reply:
Done.